Source/TCPConnection.swift

```diff
@@ -137,32 +137,28 @@ public final class TCPConnection: Connection {
             tcprecvuntil(socket, $0.baseAddress, $0.count, delimiter, delimiter.utf8.count, deadline.int64milliseconds)
         }
 
-        try TCPError.assertNoReceiveErrorWithData(data, bytesProcessed: bytesProcessed)
+        try TCPError.assertNoReceiveError(withData: data, bytesProcessed: bytesProcessed)
         return Data(data.prefix(bytesProcessed))
     }
 
-    public func close() -> Bool {
+    public func close() throws {
         guard let socket = self.socket else {
             closed = true
-            return true
+            throw TCPError.closedSocket(description: "Connection has not been initialized. You must first open to the connection.")
         }
 
         if closed {
-            return false
+            throw ClosableError.alreadyClosed
         }
 
         closed = true
         tcpclose(socket)
-        return true
     }
 
     private func getSocket() throws -> tcpsock {
         guard let socket = self.socket else {
             throw TCPError.closedSocket(description: "Connection has not been initialized. You must first open to the connection.")
         }
-        if socket == nil {
-            throw TCPError.closedSocket(description: "Connection has not been initialized. You must first open to the connection.")
-        }
```

Comment:
When I added this the socket variable would be defined but it would point to 0x00000000 (nil) and pass the guard but not the nil check.